lua/astrocommunity/pack/haskell/init.lua

```diff
@@ -14,19 +14,27 @@ return {
     "mrcjkb/haskell-tools.nvim",
     dependencies = {
       "nvim-lua/plenary.nvim",
-      { "nvim-telescope/telescope.nvim", optional = true }, -- Optional
+      { "nvim-telescope/telescope.nvim", optional = true },
+      { "mfussenegger/nvim-dap", optional = true },
     },
     branch = "2.x.x", -- Recommended
-    init = function() -- Optional, see Advanced configuration
-      astronvim.lsp.skip_setup = utils.list_insert_unique(astronvim.lsp.skip_setup, "hls")
-      vim.g.haskell_tools = {
+    init = function() astronvim.lsp.skip_setup = utils.list_insert_unique(astronvim.lsp.skip_setup, "hls") end,
+    opts = function(_, opts)
+      -- this allows users to pass opts through an opts table in community.lua
+      opts = opts or {}
+      local defaults = {
         hls = {
-          on_attach = function(client, bufnr, ht)
-            require("astronvim.utils.lsp").on_attach(client, bufnr)
-            ht.dap.discover_configurations(bufnr)
-          end,
+          on_attach = function(client, bufnr, _) require("astronvim.utils.lsp").on_attach(client, bufnr) end,
         },
       }
+      opts = vim.tbl_deep_extend("keep", opts, defaults)
+    end,
+    config = function(_, opts)
+      -- haskell-tools reads this config the first time `require('haskell-tools')` is called,
+      -- which is typically when opening a Haskell or Cabal file, or when the
+      -- user executes one of the plugin's commands.
+      -- NOTE: If vim.g.haskell_tools has already been set, its options take precedence.
+      vim.g.haskell_tools = vim.tbl_deep_extend("keep", vim.g.haskell_tools or {}, opts)
```

Comment:
Not sure if it would be cleaner, but I guess you could also set `vim.g.haskell_tools` in the `opts` function instead?

Reply:
setting it here is correct. Moving it to the `opts` function will break the user extensibility